libtenzir/src/tql2/plugin.cpp

```diff
@@ -23,28 +25,77 @@ auto function_use::evaluator::operator()(const ast::expression& expr) const
 
 auto aggregation_plugin::make_function(invocation inv, session ctx) const
   -> failure_or<function_ptr> {
-  // TODO: Consider making this pure-virtual or provide a default implementation.
-  diagnostic::error("this function can only be used as an aggregation function")
-    .primary(inv.call.fn)
-    .emit(ctx);
-  return failure::promise();
+  if (inv.call.args.empty()) {
+    diagnostic::error("aggregation functions require at least one argument")
+      .primary(inv.call)
+      .emit(ctx);
+    return failure::promise();
```

Comment:
Minor detail, but this is not true. `count()` has not argument, it just doesn't work in a normal expression without `summarize`. Maybe we can slightly tweak the message here?

Reply:
I reworded this and also put in a hint to use `summarize`.